src/Domain/Bootstrap.php

```diff
@@ -156,6 +156,13 @@ protected function remove_core_blocks() {
 	 */
 	protected function define_feature_flag() {
 		$woo_options = parse_ini_file( __DIR__ . '/../../blocks.ini' );
-		define( 'WOOCOMMERCE_BLOCKS_PHASE', $woo_options['woocommerce_blocks_phase'] );
+		$flag        = 'stable';
+		if ( is_array( $woo_options ) && (
+				'experimental' === $woo_options['woocommerce_blocks_phase'] ||
+				'stable' === $woo_options['woocommerce_blocks_phase']
+		) ) {
+			$flag = sanitize_key( $woo_options['woocommerce_blocks_phase'] );
+		}
+		define( 'WOOCOMMERCE_BLOCKS_PHASE', $flag );
```

Comment:
If `$flag` defaults to stable, then you only have the check for `experimental` to override right?  Also, no need to run though `sanitize_key` if you are doing a whitelist.  So you could probably simplify this by doing:

```php
$flag = is_array( $woo_options ) && 'experimental' === $woo_options['woocommerce_blocks_phase'] ? 'experimental' : 'stable';
define( 'WOOCOMMERCE_BLOCKS_PHASE', $flag );
```

Reply:
I applied your fix and added the env to travis, I did a check first for the env variable than to the blocks.ini, it should be safe for any environment.

```php
if ( getenv( 'WOOCOMMERCE_BLOCKS_PHASE' ) ) {
	$flag = getenv( 'WOOCOMMERCE_BLOCKS_PHASE' );
} else {
	$woo_options = parse_ini_file( __DIR__ . '/../../blocks.ini' );
	$flag        = is_array( $woo_options ) && 'experimental' === $woo_options['woocommerce_blocks_phase'] ? 'experimental' : 'stable';
}
```